rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -69,17 +82,39 @@ namespace reinforcement_learning { namespace model_management {
     }
   }
 
+  int restapi_data_transport::get_data_info_using_get(http_response& response, api_status* status) {
+      http_request request(methods::GET);
+      request.headers() = _header;
+      // Build request URI and start the request.
+      auto request_task = _httpcli->request(request).then([&](http_response get_response) {
+          if (get_response.status_code() != 200)
+          {
+              RETURN_ERROR_ARG(_trace, status, http_bad_status_code, "Found: ", response.status_code(), _httpcli->get_url());
+          }
+          response = get_response;
+          return error_code::success;
+      });
+      try {
+          return request_task.get();
+      }
+      catch (const std::exception& e) {
+          RETURN_ERROR_LS(_trace, status, exception_during_http_req) << e.what() << "\n URL: " << _httpcli->get_url();
+      }

```

Comment:
This seems to be mostly duplicated with get_data_info. Why not factor out into a `make_request_foo` and factor out the HTTP method being used?

Reply:
updated logi